src/runtime/disco/ccl/ccl.cc

```diff
@@ -16,9 +16,48 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+#ifndef TVM_NCCL_RCCL_SWITCH
+#define TVM_NCCL_RCCL_SWITCH 0  // 0: NCCL, 1: RCCL
+#endif
+
+#if TVM_NCCL_RCCL_SWITCH == 0
 #include <cuda_runtime_api.h>
-#include <dlpack/dlpack.h>
 #include <nccl.h>
+
+#include "../../cuda/cuda_common.h"
+
+using runtimeStream_t = cudaStream_t;
+
+#define runtimeCall CUDA_CALL
+#define SetDevice cudaSetDevice
+#define StreamCreate cudaStreamCreate
+#define StreamSync cudaStreamSynchronize
+#define StreamDestroy cudaStreamDestroy
+#define CommDestroy ncclCommDestroy
+#define RuntimeDeviceType DLDeviceType::kDLCUDA
+#define runtimeString "cuda"
+#define CCLString "nccl"
```

Comment:
if possible, avoid `#define` for constant values, and we should follow google c++ naming style:

```suggestion
#define TVM_DISCO_DEVICE_CALL CUDA_CALL
#define TVM_DISCO_DEVICE_SET_DEVICE cudaSetDevice
#define TVM_DISCO_DEVICE_STREAM_CREATE cudaStreamCreate
#define TVM_DISCO_DEVICE_STREAM_SYNC cudaStreamSynchronize
#define TVM_DISCO_DEVICE_STREAM_DESTROY cudaStreamDestroy
#define TVM_DISCO_DEVICE_NAME "cuda"
#define TVM_DISCO_CCL_DESTROY ncclCommDestroy
#define TVM_DISCO_CCL_NAME "nccl"
const constexpr DLDeviceType = DLDeviceType::kDLCUDA;
```